### DISCUSSION TITLE
Updates for AVS+?

### DISCUSSION THREAD

**Discussion by LeXXuz** [WolframRhodium/VapourSynth-BM3DCUDA#20] [General]
Hi there!
Any chance for updates to the AVS+ version?
Sadly I can't run Vapour Synth as I need many other filters and scripts that are not ported to VS (yet).

**Comment by WolframRhodium** [comment]
Hi.

I am waiting nv for a feedback on my bug report about incorrect vbm3d outputs on 9 and 10 series gpus (using a new cuda, which the avs+ version is not compiled with). Then the AVS+ version will be updated along with the VS version.

**Reply by LeXXuz** [reply]
Thanks for the info. :)

**Comment by WolframRhodium** [comment]
Done [test10.zip](https://github.com/WolframRhodium/VapourSynth-BM3DCUDA/files/10505990/BM3DCUDA_AVS-test10.zip).

**Reply by LeXXuz** [reply]
Awesome! I'll give it a go asap. Thank you very much!

**Comment by Galaxoid-333** [comment]
Tell me which version of AVS BM3D_CPU can I use on a processor with AVX (i7-3770)? On all versions that I tried, the error is reported:
BM3D_CPU: requires AVX2-capable cpu

**Reply by WolframRhodium** [reply]
None, AVX is not supported by design because of the use of some AVX2 instructions.

**Reply by Galaxoid-333** [reply]
> None, AVX is not supported by design because of the use of some AVX2 instructions.

It's clear. It's just that some test releases say about AVX. Also, vapoursynth version works fine on my processor.

**Reply by WolframRhodium** [reply]
AVX is for CUDA implementation. CPU version should never support non-AVX2 processors. https://github.com/HomeOfVapourSynthEvolution/VapourSynth-BM3D is available on VapourSynth but I am not the maintainer.
